### ISSUE TITLE
Allow negation in "for ! domain <domain> virtual <virtuals>"

### ISSUE THREAD

**Issue by jdelic** [OpenSMTPD/OpenSMTPD#751]
Currently `accept ... for domain <domain> virtual <virtuals>` can not be used with `relay via` (it produces an error message saying so).

What I would like to do is send email through amavisd, *but also* first deny any email to non-existent recipients. The solution *could be*

```
listen on (ip) port 25 tag UNFILTERED
accept tagged UNFILTERED from any for domain <domains> virtual <validrecipients> relay via "smtp://localhost:10025"
listen on localhost port 10026 tag FILTERED
```

but as mentioned above that produces an error message because `relay via` doesn't work like that.

So I tried *rejecting* the message first if it is addressed to a non-existent user:

```
listen on (ip) port 25 tag UNFILTERED
reject tagged UNFILTERED from any for ! domain <domains> virtual <validrecipients>
accept tagged UNFILTERED from any for any relay via "smtp://localhost:10025"
listen on localhost port 10026 tag FILTERED
```

but that results in 

```
$ smtpd -d -v
/etc/smtpd.conf:44: syntax error
```

That seems like a bug, as the [relevant documentation](https://www.opensmtpd.org/faq/rules.html) at least reads like it should work

Is there a workaround that I can use to achieve my goal of OpenSMTPD first denying any incoming email not addressed (via RCPT TO) to a valid recipient, but then relaying it to an external scanner?

**Comment by jdelic** [comment]
Since I'm banging my head against OpenSMTPD's rule system for the last couple of days... I just want to mention this if anyone ends up here via Google:

`relay via` seems to work fine with `recipients <valid_rcptto_table>`, so that kind of works as a work-around if you can compile a list of valid recipients by moving the table to a database, for example.

**Comment by jdelic** [comment]
Ok, here is a `relay via` rule that works and does what I want using another table:

```
accept from any for domain <validomains> recipient <validrecipientemails> for relay via "smtp://127.0.0.1:10026"
```

This makes sense after reviewing the code and reading the existing documentation carefully, since `virtual` is a mapping table and `recipient` is a `mailaddr` table (`query_mailaddr` for database configs). That means: One is a *dictionary lookup*, the other is a *list*.

So in the end, after figuring this out I think that this too deserves some added documentation in the man page.

Basically two documentation updates would be good:
  * change the documentation to make clear that `for domain ... virtual` can't be used with `reject`
  * change the documentation to point out that `accept ... relay via` can't be used with `for domain ... virtual` either and that the desired process can likely be achieved with `recipient <table>`.

**Comment by jdelic** [comment]
I will file individual tickets for the points mentioned in my previous comment, but this can be closed now.
